sdk/textanalytics/Azure.AI.TextAnalytics/tests/MockTests/Legacy/TextAnalyticsClientMockTests.cs

```diff
@@ -333,96 +332,6 @@ public async Task AnalyzeSentimentNullText()
             Assert.AreEqual(string.Empty, response.Sentences.FirstOrDefault().Text);
         }
 
-        [Test]
-        [Ignore("Figure out why it fails")]
-        public void AnalyzeSentimentNotSupportedSentenceSentiment()
-        {
-            using var stream = new MemoryStream(Encoding.UTF8.GetBytes(@"
-                {
-                    ""documents"": [
-                        {
-                            ""id"": ""1"",
-                            ""sentiment"": ""neutral"",
-                            ""confidenceScores"": {
-                                ""positive"": 0.1,
-                                ""neutral"": 0.88,
-                                ""negative"": 0.02
-                            },
-                            ""sentences"": [
-                                {
-                                ""sentiment"": ""confusion"",
```

Comment:
This is an interesting case. In the legacy code, `SentenceSentiment.Sentiment` (previously called `SentenceSentimentInternal`) is of type `string`. We manually modified it [here](https://github.com/Azure/azure-sdk-for-net/blob/Azure.AI.TextAnalytics_5.2.0-beta.2/sdk/textanalytics/Azure.AI.TextAnalytics/src/Internal/SentenceSentimentInternal.cs#L15). 

I don't remember why, but it seems associated to [this line of code](https://github.com/Azure/azure-sdk-for-net/blob/Azure.AI.TextAnalytics_5.2.0-beta.2/sdk/textanalytics/Azure.AI.TextAnalytics/src/SentenceSentiment.cs#L37):
```
Sentiment = (TextSentiment)Enum.Parse(typeof(TextSentiment), sentenceSentiment.Sentiment, ignoreCase: true);
```

I think we wanted to just have one enum for the sentiments and we thought it was easier from string than doing a conversion.

This test is verifying that if the service sends a value that is not valid, our parsing `enum` throws an `ArgumentException`.

With the [new implementation](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/textanalytics/Azure.AI.TextAnalytics.Legacy.Shared/src/Generated/Models/SentenceSentiment.cs#L67), where we leave the `SentenceSentiment.Sentiment` as `SentenceSentimentValue` our deserialization [code fails](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/textanalytics/Azure.AI.TextAnalytics.Legacy.Shared/src/Generated/Models/SentenceSentimentValue.Serialization.cs#L19) as `confusion` is not a valid value.

Given that this is a change in behavior, the question is, what do we do?

What I am proposing is to delete the tests altogether and leave behavior as is. This is a validation we are doing in case the service fails (which we haven't seen), but we also introduced it by making the type a string. If we follow strictly what the REST API says and what we've seen from the service behavior, this is a very very unlikely scenario.

If we want to go back to the previous behavior, I will go ahead and make the modifications in the Legacy.Shared project

@jsquire what do you think?

Reply:
Given that the legacy service is stable and not under active development, if an unknown value gets returned then it would be a service problem.   Since this wasn't an extensible enum to start with, my assumption is that the service is expected to return only one of the known set for that service version.

Based on that assumption, I agree with your assessment.   Having the enum directly in the generated model and allowing it to validate the service return makes sense.   Killing the test makes sense to me.